exp/metric-gen/metric/marker_info.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2023 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package metric
+
+import (
+	"sigs.k8s.io/controller-tools/pkg/markers"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kube-state-metrics/v2/pkg/customresourcestate"
+)
+
+const (
+	// InfoMarkerName is a marker for defining metric definitions.
+	InfoMarkerName = "Metrics:info"
+)
+
+func init() {
+	markerDefinitions = append(
+		markerDefinitions,
+		must(markers.MakeDefinition(InfoMarkerName, markers.DescribesField, InfoMarker{})).
+			help(InfoMarker{}.help()),
+		must(markers.MakeDefinition(InfoMarkerName, markers.DescribesType, InfoMarker{})).
+			help(InfoMarker{}.help()),
+	)
+}
+
+type InfoMarker struct {
+	Name           string
+	Help           string              `marker:"help,optional"`
+	LabelsFromPath map[string]JSONPath `marker:"labelsFromPath,optional"`
+	JSONPath       JSONPath            `marker:"JSONPath,optional"`
```

Comment:
How about only using JSONPath in marker? And remove labelsFromPath and labelFromKey.

Reply:
I'd need to take a look, but I have doubts that this could translate to a configuration tackling the same use case 🤔

Reply:
Note: only using JSONPath does not work. `JSONPath` does resolve into the [path](https://github.com/kubernetes/kube-state-metrics/blob/d7d5ef27c57abbe4b02cddfa0b4a0f85e5f69e88/pkg/customresourcestate/config_metrics_types.go#L34) attribute. As addition `labelsFromPath` still makes sense to configure the [LabelsFromPath](https://github.com/kubernetes/kube-state-metrics/blob/d7d5ef27c57abbe4b02cddfa0b4a0f85e5f69e88/pkg/customresourcestate/config_metrics_types.go#L32)